Terminal selection and focus

larskarlitski opened the pull request:
Fix terminal selection issues. The problem is that term.js blinks the cursor by constantly removing and inserting a span, which clears the selection of the cursor line. Blink the cursor with a css animation instead. This makes copy and paste (also in curses applications) with the right-click menu at least possible.

Also fix the line-height to make selections look better.

Depends on term.js-cockpit 0.0.9:
- [x] https://github.com/cockpit-project/term.js/pull/2
- [x] https://github.com/cockpit-project/term.js/pull/4

larskarlitski commented:
Note: Tests are failing because the term.js dependency isn't updated on npm yet. That pull requests needs a review first.

mvollmer commented:
https://github.com/cockpit-project/term.js/pull/2 is merged.  Do we need anything else?  `npm` pulls from GitHub, right?

larskarlitski commented:
> cockpit-project/term.js#2 is merged. Do we need anything else? npm pulls from GitHub, right?

No, it pulls from npmjs.org, but I've just uploaded it. Thanks!

mvollmer commented:
The tests fail with this error:
```
Page error: TypeError: undefined is not a function (evaluating 'y.showCursor()')
```

larskarlitski commented:
Ya, I removed `showCursor()` from our term.js fork. Sorry, I thought it was only used internally.

There was never a need to call this function though, as we're setting `term.cursorHidden = false` anyway. I've removed it and added an explanation to the commit message.

larskarlitski commented:
Hm, apparently `kubernetes-container-terminal` used `showCursor()` as well. I didn't catch that, because its in a different repository (yay for testing dependants).

I've reverted the removal of `showCursor()` in https://github.com/cockpit-project/term.js/pull/4. Once that is in, the tests *should* pass.

larskarlitski commented:
Thanks @martinpitt for having a look. Published `term.js-cockpit` and retriggered the tests.

martinpitt commented on package.json:
```diff
@@ -28,7 +28,7 @@
     "registry-image-widgets": "0.0.13",
     "requirejs": "2.1.22",
-    "term.js-cockpit": "0.0.8"
+    "term.js-cockpit": "0.0.9"
```
Doesn't this need to be bumped to 0.0.10 too?

martinpitt replied:
Ah, I suppose not, as cockpit itself doesn't use the `showCursor()` call? But I wonder why this was blocked on the term.js 0.0.10 PR then.

martinpitt commented:
Force-pushed with these changes:
 * Bump term.js-cockpit dependency (also in commit message)
 * Add Closes # to commit message
 * Capitalize commit messages
